option java_package = "org.kegbot.proto";
import "models.proto";

// Common

message Paging {
  optional uint32 total = 1;
  optional uint32 limit = 2;
  optional uint32 pos = 3;
}

// Responses

message DrinkSet {
  repeated Drink drinks = 1;
  optional Paging paging = 2;
}

message KegSet {
  repeated Keg kegs = 1;
  optional Paging paging = 2;
}

message SessionSet {
  repeated Session sessions = 1;
  optional Paging paging = 2;
}

message SystemEventSet {
  repeated SystemEvent events = 1;
  optional Paging paging = 2;
}

message SystemEventDetailSet {
  repeated SystemEventDetail events = 1;
  optional Paging paging = 2;
}

message SystemEventHtmlSet {
  repeated SystemEventHtml events = 1;
  optional Paging paging = 2;
}

message SoundEventSet {
  repeated SoundEvent events = 1;
  optional Paging paging = 2;
}

message TapDetailSet {
  repeated TapDetail taps = 1;
  optional Paging paging = 2;
}

message DrinkDetailHtmlSet {
  repeated DrinkDetailHtml drinks = 1;
  optional Paging paging = 2;
}

message ThermoSensorSet {
  repeated ThermoSensor sensors = 1;
  optional Paging paging = 2;
}

message ThermoLogSet {
  repeated ThermoLog logs = 1;
  optional Paging paging = 2;
}

message TapDetail {
  required KegTap tap = 1;
  optional Keg keg = 2;
  optional BeerType beverage = 3;
  optional Brewer brewer = 4;
}

message DrinkDetail {
  required Drink drink = 1;
  optional User user = 2;
  optional Keg keg = 3;
  optional Session session = 4;
}

message SessionDetail {
  required Session session = 1;
  optional string stats = 2;
  repeated Keg kegs = 3;
}

message KegDetail {
  required Keg keg = 1;
  optional BeerType type = 2;
  optional KegSize size = 3;
  repeated Drink drinks = 4;
  repeated Session sessions = 5;
}

message UserDetail {
  required User user = 1;
}

message SystemEventDetail {
  required SystemEvent event = 1;
  optional Image image = 2;
}

message SystemEventHtml {
  required string id = 1;
  optional string html = 2;
}

message ThermoSensorDetail {
  required ThermoSensor sensor = 1;
  optional float last_temp = 2;
  optional string last_time = 3;
}

message DrinkDetailHtml {
  required string id = 1;
  required string box_html = 2;
}

service KegwebApi {
  //rpc getLastDrinks (NullRequest) returns LastDrinkResponse;
}
